ps_facetedsearch.php

```diff
@@ -1672,6 +1681,49 @@ public function getProductByFilters(
         );
     }
 
+    /**
+     * EDVINAS - Include sql ORDER BY string to be able sort products: inStock, Allowed oosp, outOfStock
+     *
+     * @param array $orderyBy - way of order
+     *
+     * @return string sql
+     */
+    public static function sqlOrderByOOSP($orderyBy)
+    {
+        if (Configuration::get('PS_LAYERED_FILTER_SHOW_OUT_OF_STOCK_LAST')
+         && $orderyBy == 'p.price') {
+            return 'stock.quantity DESC, FIELD(stock.out_of_stock, 1, 2, 0) ASC, ';
+        }
+    }
+
+    /**
+     * EDVINAS - Modify SQL of getProducts method.
+     *
+     * @param array $result
+     *
+     * @return array products
+     */
+    public static function modifyGetProducts($array)
+    {
+        $outOfStockProducts = array();
+        $outOfStockOOPSProducts = array();
+
+        foreach($array as $subKey => $subArray){
+            // SPLIT out_of_stock OOPS products
+            if(Product::isAvailableWhenOutOfStock($subArray['out_of_stock']) == 1 && $subArray['quantity'] <= 0) {
+                $outOfStockOOPSProducts[] = $array[$subKey];
+                unset($array[$subKey]);
+            }
+
+            // SPLIT out_of_stock products
+            if(Product::isAvailableWhenOutOfStock($subArray['out_of_stock']) == 0 && $subArray['quantity'] <= 0) {
+                $outOfStockProducts[] = $array[$subKey];
+                unset($array[$subKey]);
+            }
```

Comment:
to avoid duplicated code:

```php
        foreach($array as $subKey => $subArray) {
            if ($subArray['quantity'] > 0) {
                continue;
            }

            if (Product::isAvailableWhenOutOfStock($subArray['out_of_stock']) == 1) {
                // SPLIT out_of_stock OOPS products
                $outOfStockOOPSProducts[] = $array[$subKey];
            } else {
                // SPLIT out_of_stock products
                $outOfStockProducts[] = $array[$subKey];
            }
            unset($array[$subKey]);
        }
```